Sources/Logto/LogtoUtilities.swift

```diff
@@ -29,20 +30,65 @@ public enum LogtoUtilities {
     /// Decode ID Token claims WITHOUT validation.
     /// - Parameter token: The JWT to decode
     /// - Returns: A set of ID Token claims
-    static func decodeIdToken(_ token: String) throws -> IdTokenClaims {
+    static func decodeIdToken(_ idToken: String) throws -> IdTokenClaims {
         let decoder = JSONDecoder()
         decoder.keyDecodingStrategy = .convertFromSnakeCase
 
-        let segments = token.split(separator: ".")
+        let segments = idToken.split(separator: ".")
 
         guard let payload = segments[safe: 1] else {
-            throw LogtoErrors.Decode.noPayloadFound
+            throw LogtoErrors.Decoding.noPayloadFound
         }
 
         guard let decoded = String.fromUrlSafeBase64(string: String(payload)) else {
-            throw LogtoErrors.Decode.invalidUrlSafeBase64Encoding
+            throw LogtoErrors.Decoding.invalidUrlSafeBase64Encoding
         }
 
         return try decoder.decode(IdTokenClaims.self, from: Data(decoded.utf8))
     }
+
+    /// Verify the give ID Token:
+    /// * One of the JWKs matches the token.
+    /// * Issuer matches token payload `iss`.
+    /// * Client ID matches token payload `aud`.
+    /// * The token is not expired.
+    /// * The token is issued in +/- 1min.
+    static func verifyIdToken(
+        _ idToken: String,
+        issuer: String,
+        clientId: String,
+        publicKeys: [RSAPublicKey],
+        forTimeInterval: TimeInterval = Date().timeIntervalSince1970
+    ) throws {
+        if publicKeys.isEmpty {
+            throw LogtoErrors.Verification.missingJwk
+        }
+
+        // Public key verification
+        let jws = try JWS(compactSerialization: idToken)
+        guard publicKeys
+            .compactMap({ try? $0.converted(to: SecKey.self) })
+            .compactMap({ Verifier(verifyingAlgorithm: .RS256, key: $0) })
+            .contains(where: {
+                (try? jws.validate(using: $0)) != nil ? true : false
+            })
+        else {
+            throw LogtoErrors.Verification.noPublicKeyMatched
+        }
+
+        // Claims verification
+        let claims = try decodeIdToken(idToken)
+        guard claims.iss == issuer else {
+            throw LogtoErrors.Verification.valueMismatch(field: .issuer)
+        }
+        guard claims.aud == clientId else {
+            throw LogtoErrors.Verification.valueMismatch(field: .audience)
+        }
+        guard claims.exp > Int64(forTimeInterval / 1000) else {
```

Comment:
nit: Since these are not covered by our SDK convention. We'd better align whether to verify the iat and exp cross all SDKs. cc: @logto-io/eng  @xiaoyijun

Reply:
it has been covered in [this page](https://www.notion.so/silverhand/verifyIdToken-b0aadab6258a4083b9024a0949994c4f)

Reply:
I prefer to verify `iat` and `exp`,
and the valid time range need to be further discussed.

Reply:
Got it, miss that part